### PR TITLE
[Improvement](auth)(step-2) add ranger authorizer for hms catalog

### PR DESCRIPTION
**Pull request by Yulei-Yang**
# Proposed changes

finish debug of hms ranger plugin

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java [hunk 1]
```diff
@@ -33,33 +34,49 @@
...
     }
 
     private RangerAccessRequestImpl createRequest(UserIdentity currentUser, HiveAccessType accessType) {
         RangerAccessRequestImpl request = new RangerAccessRequestImpl();
-        request.setUser(currentUser.getQualifiedUser());
-        request.setUserRoles(currentUser.getRoles());
+        // currentUser.getQualifiedUser() is as of form: default_cluster:user1, only use `user1`
+        String[] userArray = currentUser.getQualifiedUser().split(":");
+        request.setUser(userArray[1]);
+        request.setClusterName(userArray[0]);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:65]: You can use `ClusterNamespace.getNameFromFullName(xxx)` directly.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java [hunk 2]
```diff
@@ -33,33 +34,49 @@
...
+        Set<String> roles = new HashSet<>();
+        for (String role : currentUser.getRoles()) {
+            // some roles are as of form: default_role_rbac_test@%, only use `default_role_rbac_test`
+            roles.add(role.split("@")[0]);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:69]: Actually, `default_role_rbac_test@%` is the real name of a user.
Each user will have an implicit default role, which name is exactly same as username(`default_role_rbac_test@%`).

And you can add new roles to user.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java [hunk 3]
```diff
@@ -33,33 +34,49 @@
...
+        }
+        request.setUserRoles(roles);
         request.setAction(accessType.name());
         if (accessType == HiveAccessType.USE) {
             request.setAccessType(RangerPolicyEngine.ANY_ACCESS);
         } else {
             request.setAccessType(accessType.name().toLowerCase());
         }
         request.setClientIPAddress(currentUser.getHost());
+        request.setClusterType(CLIENT_TYPE_DORIS);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:79]: `CLIENT_TYPE_DORIS`, what is this for? Does user need to set it somewhere in Ranger?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java [hunk 4]
```diff
@@ -99,20 +112,20 @@ private boolean checkPrivilege(UserIdentity currentUser, HiveAccessType accessTy
         request.setResource(resource);
 
         RangerAccessResult result = hivePlugin.isAccessAllowed(request, auditHandler);
-        auditHandler.flushAudit();
 
         if (result == null) {
-            LOG.warn(String.format("Error getting authorizer result, please check your ranger config. Request: %s",
-                    request));
+            LOG.warn(String.format("Error getting authorizer result, please check your ranger config. Make sure "
+                    + "ranger policy engine is initialized. Request: %s", request));
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:118]: Maybe we should tell user how to check Ranger's log, eg, add `log4j.xml` in `fe/conf`, in our document.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java [hunk 5]
```diff
@@ -152,27 +165,30 @@ public HiveAccessType convertToAccessType(PrivPredicate predicate) {
...
         } else if (predicate == PrivPredicate.DROP) {
             return HiveAccessType.DROP;
-        } else if (predicate == PrivPredicate.SELECT) {
-            return HiveAccessType.SELECT;
         } else {
             return HiveAccessType.NONE;
         }
     }
 
+    private String removeDefaultClusterName(String nameWithDefaultCluster) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:173]: See `ClusterNamespace.getNameFromFullName(xxx)`

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 1 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:65]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 5 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:173]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 2 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 3 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:83]: used for ranger audit, user does not need to set it

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 4 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:122]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java (hunk 3 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java:83]: it's a constant defined in RangerHiveAccessController

**Comment by Yulei-Yang** [comment]
run beut

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run beut

**Comment by Yulei-Yang** [comment]
run beut

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**